src/amo/components/Collection/index.js

```diff
@@ -267,7 +267,7 @@ export class CollectionBase extends React.Component<InternalProps> {
     const newCollectionPage = computeNewCollectionPage(collection);
 
     if (page !== newCollectionPage) {
-      page = newCollectionPage;
+      page = Number(newCollectionPage);
```

Comment:
I think this is ok to leave as string type. If you look at the original type of the page var, you can see it's a string there.